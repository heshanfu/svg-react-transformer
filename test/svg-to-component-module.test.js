'use strict';

const fs = require('fs');
const path = require('path');
const svgToComponentModule = require('../lib/svg-to-component-module');

const getFixture = name => {
  return fs.readFileSync(
    path.join(__dirname, 'fixtures', `${name}.svg`),
    'utf8'
  );
};

describe('svgToComponentModule', () => {
  test('works with an airplane', () => {
    return svgToComponentModule(getFixture('airplane')).then(result => {
      expect(result).toMatchSnapshot();
    });
  });

  test('works with an apple', () => {
    return svgToComponentModule(getFixture('apple')).then(result => {
      expect(result).toMatchSnapshot();
    });
  });

  test('passes SVGO plugins', () => {
    const options = {
      svgoPlugins: [{ removeDoctype: true }, { removeDimensions: true }]
    };
    return svgToComponentModule(getFixture('apple'), options).then(result => {
      expect(result).toMatchSnapshot();
    });
  });

  test('options.name', () => {
    const options = {
      name: 'apple'
    };
    return svgToComponentModule(getFixture('apple'), options).then(result => {
      expect(result).toMatchSnapshot();
    });
  });

  test('options.propTypes', () => {
    const options = {
      propTypes: {
        width: 'PropTypes.number.isRequired',
        height: 'PropTypes.number.isRequired'
      }
    };
    return svgToComponentModule(getFixture('apple'), options).then(result => {
      expect(result).toMatchSnapshot();
    });
  });

  test('options.template', () => {
    const options = {
      name: 'Fakery',
      propTypes: { width: 'PropTypes.number' },
      template: data => {
        return JSON.stringify(data, null, 2);
      }
    };
    return svgToComponentModule(getFixture('apple'), options).then(result => {
      expect(result).toMatchSnapshot();
    });
  });
});
