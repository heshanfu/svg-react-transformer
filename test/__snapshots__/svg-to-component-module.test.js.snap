// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`svgToComponentModule options.name 1`] = `
"\\"use strict\\";
const React = require(\\"react\\");

class Apple extends React.PureComponent {
  render() {
    return (
      <svg {...this.props} height=\\"30\\" viewBox=\\"0 0 18 18\\" width=\\"30\\">
        <path d=\\"M3.4 9.2c.1-1.1.4-2.3 1.5-3.2.9-.7 1.8-.8 2.9-.5 1.4.5 1 .5 2.4 0 1.5-.6 3 0 3.7.8.1.2.2.3 0 .4-1.8 1.3-1.6 3.9.3 5 .2.1.3.2.2.4-.4 1-1 1.9-1.7 2.7-.5.5-1.1.7-1.8.4-.1 0-.3-.1-.4-.1-.9-.1-1.7-.1-2.5.2-.3.1-.5.2-.8.3-.4.1-.8-.1-1.1-.4-.7-.5-1.1-1.2-1.5-1.9-.7-1.1-1.2-2.7-1.2-4.1\\" />
        <path d=\\"M11.6 2.5c0 1.2-1 2.4-2 2.7-.6.1-.8 0-.7-.6.2-1.2 1.2-2.2 2.5-2.5.2 0 .2 0 .2.2v.2\\" />
      </svg>
    );
  }
}

module.exports = Apple;
"
`;

exports[`svgToComponentModule options.propTypes 1`] = `
"\\"use strict\\";
const React = require(\\"react\\");
const PropTypes = require(\\"prop-types\\");

class SvgComponent extends React.PureComponent {
  render() {
    return (
      <svg {...this.props} height=\\"30\\" viewBox=\\"0 0 18 18\\" width=\\"30\\">
        <path d=\\"M3.4 9.2c.1-1.1.4-2.3 1.5-3.2.9-.7 1.8-.8 2.9-.5 1.4.5 1 .5 2.4 0 1.5-.6 3 0 3.7.8.1.2.2.3 0 .4-1.8 1.3-1.6 3.9.3 5 .2.1.3.2.2.4-.4 1-1 1.9-1.7 2.7-.5.5-1.1.7-1.8.4-.1 0-.3-.1-.4-.1-.9-.1-1.7-.1-2.5.2-.3.1-.5.2-.8.3-.4.1-.8-.1-1.1-.4-.7-.5-1.1-1.2-1.5-1.9-.7-1.1-1.2-2.7-1.2-4.1\\" />
        <path d=\\"M11.6 2.5c0 1.2-1 2.4-2 2.7-.6.1-.8 0-.7-.6.2-1.2 1.2-2.2 2.5-2.5.2 0 .2 0 .2.2v.2\\" />
      </svg>
    );
  }
}

SvgComponent.propTypes = {
  width: PropTypes.number.isRequired,
  height: PropTypes.number.isRequired
};

module.exports = SvgComponent;
"
`;

exports[`svgToComponentModule options.template 1`] = `
"{
  \\"svgJsx\\": \\"<svg\\\\n  height=\\\\\\"30\\\\\\"\\\\n  viewBox=\\\\\\"0 0 18 18\\\\\\"\\\\n  width=\\\\\\"30\\\\\\"\\\\n>\\\\n  <path\\\\n    d=\\\\\\"M3.4 9.2c.1-1.1.4-2.3 1.5-3.2.9-.7 1.8-.8 2.9-.5 1.4.5 1 .5 2.4 0 1.5-.6 3 0 3.7.8.1.2.2.3 0 .4-1.8 1.3-1.6 3.9.3 5 .2.1.3.2.2.4-.4 1-1 1.9-1.7 2.7-.5.5-1.1.7-1.8.4-.1 0-.3-.1-.4-.1-.9-.1-1.7-.1-2.5.2-.3.1-.5.2-.8.3-.4.1-.8-.1-1.1-.4-.7-.5-1.1-1.2-1.5-1.9-.7-1.1-1.2-2.7-1.2-4.1\\\\\\"\\\\n  />\\\\n  <path\\\\n    d=\\\\\\"M11.6 2.5c0 1.2-1 2.4-2 2.7-.6.1-.8 0-.7-.6.2-1.2 1.2-2.2 2.5-2.5.2 0 .2 0 .2.2v.2\\\\\\"\\\\n  />\\\\n</svg>\\",
  \\"name\\": \\"Fakery\\",
  \\"propTypes\\": {
    \\"width\\": \\"PropTypes.number\\"
  }
}"
`;

exports[`svgToComponentModule passes SVGO plugins 1`] = `
"\\"use strict\\";
const React = require(\\"react\\");

class SvgComponent extends React.PureComponent {
  render() {
    return (
      <svg {...this.props} viewBox=\\"0 0 18 18\\">
        <path d=\\"M3.4 9.2c.1-1.1.4-2.3 1.5-3.2.9-.7 1.8-.8 2.9-.5 1.4.5 1 .5 2.4 0 1.5-.6 3 0 3.7.8.1.2.2.3 0 .4-1.8 1.3-1.6 3.9.3 5 .2.1.3.2.2.4-.4 1-1 1.9-1.7 2.7-.5.5-1.1.7-1.8.4-.1 0-.3-.1-.4-.1-.9-.1-1.7-.1-2.5.2-.3.1-.5.2-.8.3-.4.1-.8-.1-1.1-.4-.7-.5-1.1-1.2-1.5-1.9-.7-1.1-1.2-2.7-1.2-4.1\\" />
        <path d=\\"M11.6 2.5c0 1.2-1 2.4-2 2.7-.6.1-.8 0-.7-.6.2-1.2 1.2-2.2 2.5-2.5.2 0 .2 0 .2.2v.2\\" />
      </svg>
    );
  }
}

module.exports = SvgComponent;
"
`;

exports[`svgToComponentModule works with an airplane 1`] = `
"\\"use strict\\";
const React = require(\\"react\\");

class SvgComponent extends React.PureComponent {
  render() {
    return (
      <svg {...this.props} viewBox=\\"0 0 18 18\\">
        <path d=\\"M7 4l1.6 4H5.5S4.1 6 3 6h-.8L3 8l1 3h4.6L7 15h2l3.2-4H14c1 0 2-.7 2-1.5S15 8 14 8h-1.8L9 4H7z\\" />
      </svg>
    );
  }
}

module.exports = SvgComponent;
"
`;

exports[`svgToComponentModule works with an apple 1`] = `
"\\"use strict\\";
const React = require(\\"react\\");

class SvgComponent extends React.PureComponent {
  render() {
    return (
      <svg {...this.props} height=\\"30\\" viewBox=\\"0 0 18 18\\" width=\\"30\\">
        <path d=\\"M3.4 9.2c.1-1.1.4-2.3 1.5-3.2.9-.7 1.8-.8 2.9-.5 1.4.5 1 .5 2.4 0 1.5-.6 3 0 3.7.8.1.2.2.3 0 .4-1.8 1.3-1.6 3.9.3 5 .2.1.3.2.2.4-.4 1-1 1.9-1.7 2.7-.5.5-1.1.7-1.8.4-.1 0-.3-.1-.4-.1-.9-.1-1.7-.1-2.5.2-.3.1-.5.2-.8.3-.4.1-.8-.1-1.1-.4-.7-.5-1.1-1.2-1.5-1.9-.7-1.1-1.2-2.7-1.2-4.1\\" />
        <path d=\\"M11.6 2.5c0 1.2-1 2.4-2 2.7-.6.1-.8 0-.7-.6.2-1.2 1.2-2.2 2.5-2.5.2 0 .2 0 .2.2v.2\\" />
      </svg>
    );
  }
}

module.exports = SvgComponent;
"
`;
